contracts/contracts/system_contracts/kaiabridge/Bridge.sol

```diff
@@ -0,0 +1,400 @@
+    }
+
+    /// @dev Update greatest sequence
+    /// @param seq ProvisionData sequence
+    function updateGreatestConfirmedSeq(uint256 seq) internal {
+        if (greatestConfirmedSeq < seq) {
+            greatestConfirmedSeq = seq;
+        }
+    }
+
+    /// @dev See {IBridge-changeMintLockablePDT}
+    function changeMintLockablePDT(uint256 newMinLockablePDT) public override onlyGuardian {
+        emit MintLockablePDTChange(minLockablePDT, newMinLockablePDT);
+        minLockablePDT = newMinLockablePDT;
+    }
+
+    /// @dev See {IBridge-changeMaxtLockablePDT}
+    function changeMaxtLockablePDT(uint256 newMaxLockablePDT) public override onlyGuardian {
+        emit MaxtLockablePDTChange(minLockablePDT, newMaxLockablePDT);
+        maxLockablePDT = newMaxLockablePDT;
+    }
+
+    /// @dev See {IBridge-changeMaxTryTransfer}
+    function changeMaxTryTransfer(uint256 newMaxTryTransfer) public override onlyGuardian {
+        emit MaxTryTransferChange(maxTryTransfer, newMaxTryTransfer);
+        maxTryTransfer = newMaxTryTransfer;
+    }
+
+    /// @dev See {IBridge-setAddrValidation}
+    function setAddrValidation(bool onOff) public override onlyGuardian {
+        emit ChangeAddrValidation(addrValidationOn, onOff);
+        addrValidationOn = onOff;
+    }
+
+    /// @dev See {IBridge-transfer}
+    function transfer(string calldata receiver)
+        public
+        override
+        payable
+        nonReentrant
+        notPause
+    {
+        if (addrValidationOn) {
+            require(verifyAddrFNSA(receiver, false), "PDT::Bridge: Receiver address is invalid");
+        }
+        require(msg.value >= minLockablePDT, "PDT::Bridge: Locked PDT must be larger than minimum");
+        require(msg.value <= maxLockablePDT, "PDT::Bridge: Locked PDT must be less than maximum");
+        seq2BlockNum[seq] = block.number;
+        SwapRequest memory swapReq = SwapRequest({
+            seq: seq++,
+            sender: msg.sender,
+            receiver: receiver,
+            amount: msg.value
+        });
+        locked.push(swapReq);
+        emit Transfer(swapReq);
+    }
+
+    /// @dev Transfer PDT to receiver with the specified amount in the provision
+    /// @param prov ProvisionData
+    /// @param revertOnFail Make reverts if operations fails and the value is true, otherwise no make revert, but record its failure
+    function claim(ProvisionData memory prov, bool revertOnFail)
+        internal
+        nonReentrant
+        enoughPoolAmount(prov.amount)
+        returns (bool)
+    {
+        // Allocate half of gas as available gas for the fallback code
+        (bool sent, ) = prov.receiver.call{
+            value: prov.amount,
+            gas: gasleft() / 2
+        }("");
+        if (!sent) {
+            if (revertOnFail) {
+                revert("PDT::Bridge: Failed to transfer amount of provision");
+            }
+            if (transferFail[prov.seq]++ > maxTryTransfer) {
```

Comment:
This allows users to try `maxTryTransfer + 2` claims at most. Was this your intention?
If not, to remove the provision at `maxTryTransfer + 1`-th `claim` call,
```suggestion
            transferFail[prov.seq]++;
            if (transferFail[prov.seq] > maxTryTransfer) {
```

Reply:
Your suggestion is the correct one. I fixed this in another repository. I'll update all together later.